oop1-construct.js

```diff
@@ -1,81 +1,277 @@
-//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.  This string should clearly explain why classes are used in Object Oriented Programming.  The first method is called explain().  This method should print the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that are essential to building a class.  Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
+//1. Create a Class called WhyClass.  This class should have one property, and two methods.  THe property is called "purpose", and stores a string.
+// This string should clearly explain why classes are used in Object Oriented Programming.  The first method is called explain().  This method should print
+// the contents stored in the purpose 'field' of the class.  The second method is called pieces().  This 2nd method should print out the various parts that
+// are essential to building a class.  Instantiate the class, and invoke the method so the results are showing in the console once this file is run.
 
 //your code here...
+class WhyClass {
+  purpose;
+
+  constructor() {
+    this.purpose =
+      'Classes are used in OOP so you dont have to rewrite things and you can encapsulate multiple forms of data.';
+  }
```

Comment:
`constructor(purpose) { this.purpose = purpose}` Always use variables in your code so that the data can be set during instantiation, unless intend the data to be `const`